### PR TITLE
Bodies & speakers: allow deleting & updated displaying of list

### LINKED ISSUE #65: Start with speakers and bodies admin
I drafted first few wireframes, drafting speakers and bodies editing. (Sorry for the quality, I stopped paying for Balsamiq and these are stitched together in visual editor.)

There's a basic menu structure (we may activate those items one by one, signaling by coloring the links, but start with a feeling of completeness; after sign up, let's use a blank page for now).

Some of the attributes are not currently in the DB schema and need to be added — but those are simple datetime or bool values. I switched bios and descriptions for external links that describe those entities better — let's show only these in admin, but on the frontend, we should abolish internally generated bios and descriptions only after a full move to the new admin (this move is okay with the team, but it will require a good load of transitional work).

screen shot 2018-03-11 at 18 56 50

(There's no wf for list of speakers, use a similar full-data dump. Let's add search/filters and better defaults (e. g. only still existing/relevant bodies as a default) after it works.)

screen shot 2018-03-11 at 18 56 59

screen shot 2018-03-11 at 18 57 24

### PR DESCRIPTION
**Pull request by vlki**
I added also delete functionality to bodies & speakers, which seems to be last thing missing to be able to close #65

### CODE REVIEW

**Comment by vlki** [comment]
How it looks:

screen shot 2018-05-21 at 18 36 23
screen shot 2018-05-21 at 18 36 33
screen shot 2018-05-21 at 18 36 54
screen shot 2018-05-21 at 18 37 09

File: app/javascript/admin/components/Speakers.tsx [hunk 1]
```diff
@@ -92,26 +128,80 @@ export default class Bodies extends React.Component<{}, ISpeakersState> {
...
+                              Smazat
+                            </button>
+                          </div>
+
+                          <h5 style={{ marginTop: 7 }}>
                             {speaker.first_name} {speaker.last_name}
                           </h5>
 
-                          <h6>{speaker.body ? speaker.body.short_name : 'Nestraník'}</h6>
+                          {/* <h6>{speaker.body ? speaker.body.short_name : 'Nestraník'}</h6> */}
```
Comment (vaclavbohac) [inline app/javascript/admin/components/Speakers.tsx]: Forgotten code?

File: app/javascript/admin/components/Speakers.tsx [hunk 2]
```diff
@@ -68,21 +85,40 @@ export default class Bodies extends React.Component<{}, ISpeakersState> {
...
+                (s) => s.id === confirmDeleteModalSpeakerId,
+              );
+
               return (
                 <div>
-                  {props.data.speakers.map((speaker: ISpeaker) => (
+                  {confirmDeleteModalSpeaker && (
+                    <ConfirmDeleteModal
+                      message={
+                        'Opravdu chcete smazat osobu ' +
```
Comment (vaclavbohac) [inline app/javascript/admin/components/Speakers.tsx]: Can you use string interpolation?

```js
`Hello, ${world}`
```

File: app/javascript/admin/components/Speakers.tsx [hunk 3]
```diff
@@ -68,21 +85,40 @@ export default class Bodies extends React.Component<{}, ISpeakersState> {
...
+                        ' ' +
+                        confirmDeleteModalSpeaker.last_name +
+                        '?'
+                      }
+                      onCancel={this.hideConfirmDeleteModal}
+                      mutation={DeleteSpeaker}
+                      mutationProps={{
+                        variables: { id: confirmDeleteModalSpeakerId },
+                        refetchQueries: [
+                          { query: GetSpeakers, variables: { name: this.state.name } },
```
Comment (vaclavbohac) [inline app/javascript/admin/components/Speakers.tsx]: Nice! 👍 

It's much less noisy than manually updating the cache & given we're implementing admin, performance does not matter that much.

File: app/javascript/admin/components/Bodies.tsx [hunk 4]
```diff
@@ -64,45 +92,110 @@ export default class Bodies extends React.Component<{}, IBodiesState> {
...
+            const confirmDeleteModalBody = props.data.bodies.find(
+              (s) => s.id === confirmDeleteModalBodyId,
+            );
+
             return (
               <div>
+                {confirmDeleteModalBody && (
+                  <ConfirmDeleteModal
+                    message={
+                      'Opravdu chcete smazat skupinu/stranu ' + confirmDeleteModalBody.name + '?'
```
Comment (vaclavbohac) [inline app/javascript/admin/components/Bodies.tsx]: Could you use string interpolation?

File: app/graphql/mutations/delete_speaker.rb [hunk 5]
```diff
@@ -2,21 +2,21 @@
...
     begin
-      # TODO: check if it destroys also all memberships and avatar
-      Speaker.destroy(args[:id].to_i)
-      202
-    rescue
-      202
+      Speaker.destroy(id)
+      id
+    rescue ActiveRecord::RecordNotFound => e
+      raise GraphQL::ExecutionError.new(e.to_s)
```
Comment (vaclavbohac) [inline app/graphql/mutations/delete_speaker.rb]: Nice, finally we could get rid of `#olol` comment :-D

File: app/javascript/admin/components/Speakers.tsx (hunk 2 above)
Comment (vlki) [reply app/javascript/admin/components/Speakers.tsx]: Sure thing! Done.

File: app/javascript/admin/components/Speakers.tsx (hunk 3 above)
Comment (vlki) [reply app/javascript/admin/components/Speakers.tsx]: Yeah, my thoughts exactly. Thanks!

File: app/javascript/admin/components/Speakers.tsx (hunk 1 above)
Comment (vlki) [reply app/javascript/admin/components/Speakers.tsx]: Thanks for noticing that. Removed!

File: app/javascript/admin/components/Bodies.tsx (hunk 4 above)
Comment (vlki) [reply app/javascript/admin/components/Bodies.tsx]: Sure thing! Done.

**Pull request merged**